client.go

```diff
@@ -163,25 +120,41 @@ func (c *Client) defaultTimeout(ctx context.Context) time.Duration {
 
 func (c *Client) registerProviders() {
 	// register ipmitool provider
-	driverIpmitool := &ipmitool.Conn{Host: c.Auth.Host, Port: c.Auth.Port, User: c.Auth.User, Pass: c.Auth.Pass, Log: c.Logger}
-	c.Registry.Register(ipmitool.ProviderName, ipmitool.ProviderProtocol, ipmitool.Features, nil, driverIpmitool)
+	ipmiOpts := []ipmitool.Option{
+		ipmitool.WithLogger(c.Logger),
+		ipmitool.WithPort(c.providerConfig.ipmitool.Port),
+		ipmitool.WithCipherSuite(c.providerConfig.ipmitool.CipherSuite),
+		ipmitool.WithIpmitoolPath(c.providerConfig.ipmitool.IpmitoolPath),
+	}
+	if driverIpmitool, err := ipmitool.New(c.Auth.Host, c.Auth.User, c.Auth.Pass, ipmiOpts...); err == nil {
+		c.Registry.Register(ipmitool.ProviderName, ipmitool.ProviderProtocol, ipmitool.Features, nil, driverIpmitool)
+	} else {
+		c.Logger.Info("ipmitool provider not available", "error", err.Error())
+	}
 
 	// register ASRR vendorapi provider
-	driverAsrockrack, _ := asrockrack.NewWithOptions(c.Auth.Host, c.Auth.User, c.Auth.Pass, c.Logger, asrockrack.WithHTTPClient(c.httpClient))
+	asrHttpClient := *c.httpClient
```

Comment:
I'm speculating here - since `http.Client.Transport` has fields with pointers, and, would it be worth deep copying the client `Transport` with [`Transport.Clone()`](https://pkg.go.dev/net/http#Transport.Clone) ?

Reply:
hmmm....looking into this i say this from the `http.Client` docs. Regardless, i have updated the code to clone the transport.

_"The Client's Transport typically has internal state (cached TCP connections), so Clients should be reused instead of created as needed. Clients are safe for concurrent use by multiple goroutines."_

Reply:
ack, feel free to drop the change if its not helping.

Reply:
i dont know the impacts here either way. i currently have the transport.Clone in there. Which way to you lean?

Reply:
I'm fine if we go with this for now and see how it works out.